test/framework/src/main/java/org/opensearch/bootstrap/BootstrapForTesting.java

```diff
@@ -301,11 +301,15 @@ static Set<URL> parseClassPathWithSymlinks() throws Exception {
     private static Set<String> getTrustedHosts() {
         //
         try {
-            return Collections.list(NetworkInterface.getNetworkInterfaces())
+            List<String> hosts = Collections.list(NetworkInterface.getNetworkInterfaces())
                 .stream()
                 .flatMap(iface -> Collections.list(iface.getInetAddresses()).stream())
                 .map(address -> NetworkAddress.format(address))
-                .collect(Collectors.toSet());
+                .collect(Collectors.toList());
+            // 0:0:0:0:0:0:0:1 is simplified to ::1, in it test, the incoming address can be 0:0:0:0:0:0:0:1,
+            // so we should add it to trusted hosts.
+            hosts.add("0:0:0:0:0:0:0:1");
+            return Collections.unmodifiableSet(new HashSet<>(hosts));
```

Comment:
@reta, I'm a little confused. .`HighLevelRestClientCompressionIT.testCompressesRequest` always fails and throws the following exception, even if i add permission in `test-framework.policy`:
```
grant codeBase "${codebase.httpclient5}" {
  // httpclient5 makes socket connections for rest tests
  permission java.net.SocketPermission "*", "connect,resolve";
};
```

```
  2> REPRODUCE WITH: ./gradlew ':client:rest-high-level:asyncIntegTest' --tests "org.opensearch.client.HighLevelRestClientCompressionIT.testCompressesRequest" -Dtests.seed=4B4989BF515EDDE4 -Dtests.opensearch.logger.level=DEBUG -Dtests.security.manager=true -Dtests.jvm.argline="-XX:TieredStopAtLevel=1 -XX:ReservedCodeCacheSize=64m" -Dtests.locale=en-JE -Dtests.timezone=SystemV/EST5 -Druntime.java=23
  2> java.security.AccessControlException: access denied ("java.net.SocketPermission" "[0:0:0:0:0:0:0:1]:63932" "connect,resolve")
        at __randomizedtesting.SeedInfo.seed([4B4989BF515EDDE4:5D262F248C170E18]:0)
        at java.base/java.security.AccessControlContext.checkPermission(AccessControlContext.java:488)
        at java.base/java.security.AccessController.checkPermission(AccessController.java:1085)
        at java.base/java.lang.SecurityManager.checkPermission(SecurityManager.java:411)
        at java.base/java.lang.SecurityManager.checkConnect(SecurityManager.java:905)
        at org.opensearch.secure_sm.SecureSM$1.checkConnect(SecureSM.java:131)
        at java.base/java.net.Socket.connect(Socket.java:756)
        at org.apache.hc.client5.http.impl.io.DefaultHttpClientConnectionOperator.connect(DefaultHttpClientConnectionOperator.java:216)
        at org.apache.hc.client5.http.impl.io.PoolingHttpClientConnectionManager.connect(PoolingHttpClientConnectionManager.java:490)
        at org.apache.hc.client5.http.impl.classic.InternalExecRuntime.connectEndpoint(InternalExecRuntime.java:164)
        at org.apache.hc.client5.http.impl.classic.InternalExecRuntime.connectEndpoint(InternalExecRuntime.java:174)

```

Reply:
Correct, the permission looks legit, it may depend on the caller (thread  or missed privileged call)